### ISSUE TITLE
Wrong profit percentage calculation

### ISSUE THREAD

**Issue by v1z** [taniman/profit-trailer#1960]
It seems that PT lost both commissions on buy and sell plus add magic number to sell price * amount value - as a result actual profit percentage equals to 0.04% instead of 0.85% which is minimum value in settings. There is some more math on a screenshot below.
I asked discord support first immediately when it happened , but still don't receive any answer.

## ProfitTrailer info
Version: 2.0.13

Exchange:
- [x] Bittrex

Market:
- [x] BTC

Operating System:
- [x] Windows (Windows Server 2012 R2 Datacenter)

## Expected Behavior
I expect PT to sell only when profit percentage is above than I set in settings (0.85% minimum)

## Current Behavior
Profit percentage equals to 0.04%
(0.00054137 - 0.00054115) / 0.00054115 * 100 = 0.04%

![20180626191637](https://user-images.githubusercontent.com/8428903/42121838-40b58d18-7c3f-11e8-88b7-363c2e0ce3b0.png)

## Configuration
https://pastebin.com/BPjL85Cm

## Log files/console output
PT log
```2018-06-26 07:19:46 INFO ProfitTrailerService - Store last order before buy/sell orderid: 806b00be-e647-4189-bcaf-a1abb92d82d1
2018-06-26 07:19:46 INFO ProfitTrailerService - Sell order for BTC-XVG sold amount 145.502646 for price 0.000004 -- {"OrderId":"23f08135-ccb1-43b9-b2ae-85531bf8fb05","MarketName":"BTC-XVG","MarketCurrency":"XVG","BuyOrSell":"Sell","OrderType":"LIMIT","Quantity":145.50264550,"Rate":0.00000371}
2018-06-26 07:19:51 INFO ProfitTrailerService - Get order information -- {"AccountId":null,"OrderUuid":"23f08135-ccb1-43b9-b2ae-85531bf8fb05","Exchange":"BTC-XVG","Type":"LIMIT_SELL","Quantity":145.50264550,"QuantityRemaining":0.00000000,"Limit":0.00000371,"Reserved":145.50264550,"ReserveRemaining":145.50264550,"CommissionReserved":0.00000000,"CommissionReserveRemaining":0.00000000,"CommissionPaid":0.00000135,"Price":0.00054272,"PricePerUnit":0.00000372000000000000,"Opened":"2018-06-26T14:19:46.31","Closed":"2018-06-26T14:19:46.45","IsOpen":false,"Sentinel":"d4ac3f66-494c-48de-baf7-40c8c94c60ab","CancelInitiated":false,"ImmediateOrCancel":true,"IsConditional":false,"Condition":"NONE","ConditionTarget":null}
2018-06-26 07:19:51 INFO ProfitTrailerService - Check last order after buy/sell orderid:23f08135-ccb1-43b9-b2ae-85531bf8fb05, retries 0
2018-06-26 07:19:51 INFO StrategyRunnerHelper - SELLLOG: {"combinedProfit":0.0,"market":"BTC-XVG","profit":0.8513513513513598,"orderbookProfit":-0.2297297297297325,"averageCalculator":{"totalCost":0.000541150005281166200,"totalAmount":145.50264550,"totalAmountWithSold":145.50264550,"avgPrice":0.0000037000000,"avgCost":0.000538359788350000000,"firstBoughtDate":{"date":{"year":2018,"month":6,"day":26},"time":{"hour":14,"minute":19,"second":43,"nano":480000000}},"totalWeightedPrice":0.0005383597883500000000000000,"fee":0.5,"usedBuyHistory":[{"globalTradeID":0,"tradeID":"806b00be-e647-4189-bcaf-a1abb92d82d1","date":{"date":{"year":2018,"month":6,"day":26},"time":{"hour":14,"minute":19,"second":43,"nano":480000000}},"rate":"0.00000370000000000000","amount":"145.5026455","total":"5.4115E-4","fee":"0","orderNumber":"806b00be-e647-4189-bcaf-a1abb92d82d1","type":"buy","category":""}],"usedSellHistory":[]},"currentPrice":3.75E-6,"buyStrategies":[],"sellStrategies":[{"type":"SELL","name":"GAIN","entryValue":0.85,"entryValueLimit":0.0,"triggerValue":0.0,"currentValue":1.1483516483516647,"currentValuePercentage":0.0,"decimals":2,"positive":"true trailing..."}],"volume":0.0,"percChange":-0.02584,"boughtTimes":0,"originalTrailingData":{"originalTrigger":0.0,"originalWidth":0.0,"originalValue":0.85,"firstTrailTime":{"date":{"year":2018,"month":6,"day":24},"time":{"hour":16,"minute":22,"second":38,"nano":884000000}},"lastTrailTime":{"date":{"year":2018,"month":6,"day":26},"time":{"hour":14,"minute":20,"second":46,"nano":196000000}}}}
2018-06-26 07:19:51 INFO StrategyRunnerHelper - STOREDAVG: {"currencyPair":"BTC-XVG","lastBoughtDate":{"date":{"year":2018,"month":6,"day":24},"time":{"hour":15,"minute":45,"second":12,"nano":524000000}},"lastOrderId":"","lastAveragePrice":0,"boughtPrice":0,"boughtPriceOrderId":"","lastTotalAmount":0,"overridden":false}```

**Comment by v1z** [comment]
I have an eye on last sales and ALL of them have wrong profit calculation it PT monitor
![20180701183102](https://user-images.githubusercontent.com/8428903/42136076-db3cf83e-7d5d-11e8-8786-e9d46426fd31.png)
![20180701183314](https://user-images.githubusercontent.com/8428903/42136077-db5ca38c-7d5d-11e8-85c9-8f15b3613cc9.png)
![20180701183538](https://user-images.githubusercontent.com/8428903/42136078-db7d03de-7d5d-11e8-80d6-d7417dd4d3e9.png)

**Comment by taniman** [comment]
@v1z The bot stores the limit order and not the actual fill prices so yes  you will have more profit when you check on the exchange.
That is not wrong.

**Comment by taniman** [comment]
About the original issue you reported. I will need u to provide more of your logfile. Cause this what you are showing is not possible. The values do not match

**Comment by v1z** [comment]
@taniman Ok, the limit order makes sense.
About original issue here is 2 log files - one from PT and another from Bittrex, hope it will help
[ProfitTrailer (1).log](https://github.com/taniman/profit-trailer/files/2153962/ProfitTrailer.1.log)
[fullOrders (3).csv.zip](https://github.com/taniman/profit-trailer/files/2153964/fullOrders.3.csv.zip)

**Comment by taniman** [comment]
I have been investigating this issue and I see that the log file and sales log got a bit confused.
The log is showing you sold 451 at 07:19. But this amount of XVG sold was your previous sold amount on the 24th.

So the incorrect data was saved to the log file due to a mysterious issue. 
I have created a possible fix for this in 2.0.14 which I just released.

Could you check in your sales log and confirm that on the 24th you also have a sale for XVG in your sales log with 451 and a similar profit %?

I have also noticed buying but no selling in the log you sent me. If you are doing external buys, that would also explain a little of the strange/fast selling. 
Cause the bot can only clear some data once it does a new initial buy.  But if buy's are happening outside of the bot, it does not know to clear the data.

tl;dr;
Just a display issue
Possible fix to prevent this from happening again in 2.0.14
Watch out with external buys
